lisa/tests/base.py

```diff
@@ -431,9 +427,22 @@ def trace(self):
         trace = Trace(path, self.plat_info, events=self.ftrace_conf["events"])
         return trace.get_view(self.trace_window(trace))
 
-    def __init__(self, res_dir, plat_info, rtapp_profile):
-        super().__init__(res_dir, plat_info)
-        self.rtapp_profile = rtapp_profile
+    @property
+    def rtapp_profile(self):
+        """
+        Compute the RTapp profile based on ``plat_info``.
+        """
+        return self.get_rtapp_profile(self.plat_info)
+
+    @classmethod
+    @abc.abstractmethod
+    def get_rtapp_profile(cls, plat_info):
```

Comment:
I can't highlight it but it's already defined at line 582 of that file. It's the bit that surprised me from the PR message - it *is* already defined in this class, otherwise it wouldn't make a lot of sense (misfit tests would have to reimplement that logic for instance)

Reply:
That's right, I was mislead by a redundant definition in EASBehaviour probably dating from before a refactoring.

> otherwise it wouldn't make a lot of sense 
hence the commit :)

> misfit tests would have to reimplement that logic for instance
Having an abstractmethod or not is orthogonal, there are many codebases that don't use ABC at all and rely on "informal ABC" that do not enforce implementation, but where it is still expected.